bottery/patterns.py

```diff
@@ -1,21 +1,33 @@
 class BaseHandler:
-    def __init__(self, pattern=None):
+    def __init__(self, pattern=None, case_sensitive=True):
         self.pattern = pattern
+        self.case_sensitive = case_sensitive
```

Comment:
This option should probably in a `mixin`. I can see some handlers that this options wouldn't make sense, e.g. RegexHandler. We also should think in a way that we isn't necessary to implement the case sensitive checker in all handlers (we can do this later :).